cms/static/js/factories/import.js

```diff
@@ -33,39 +51,36 @@ define([
             maxChunkSize: 20 * 1000000, // 20 MB
             autoUpload: false,
             add: function(e, data) {
-                Import.clearImportDisplay();
-                Import.okayToNavigateAway = false;
+                Import.reset();
                 submitBtn.unbind('click');
+
                 file = data.files[0];
+
                 if (file.name.match(/tar\.gz$/)) {
-                    submitBtn.click(function(event){
+                    submitBtn.click(function(event) {
                         event.preventDefault();
-                        $.cookie('lastfileupload', file.name);
+
+                        Import.start(
+                            file.name,
+                            feedbackUrl.replace('fillerName', file.name)
+                        ).then(onComplete);
+
                         submitBtn.hide();
-                        Import.startUploadFeedback();
-                        data.submit().complete(function(result, textStatus, xhr) {
-                            window.onbeforeunload = null;
-                            if (xhr.status != 200) {
+                        data.submit().complete(function (result, textStatus, xhr) {
+                            if (xhr.status !== 200) {
                                 var serverMsg, errMsg, stage;
+
                                 try{
-                                    serverMsg = $.parseJSON(result.responseText);
+                                    serverMsg = $.parseJSON(result.responseText) || {};
                                 } catch (e) {
                                     return;
                                 }
-                                errMsg = serverMsg.hasOwnProperty('ErrMsg') ?  serverMsg.ErrMsg : '' ;
-                                if (serverMsg.hasOwnProperty('Stage')) {
-                                    stage = Math.abs(serverMsg.Stage);
-                                    Import.stageError(stage, defaults[stage] + errMsg);
-                                }
-                                else {
-                                    alert(gettext('Your import has failed.') + '\n\n' + errMsg);
-                                }
-                                chooseBtn.html(gettext('Choose new file')).show();
-                                bar.hide();
+
+                                errMsg = serverMsg.hasOwnProperty('ErrMsg') ? serverMsg.ErrMsg : '';
+                                stage = Math.abs(serverMsg.Stage || 0);
```

Comment:
@AkA84 the code here is not quite the same as it was - namely, if `serverMsg` didn't contain `Stage` it would alert "Your import has failed." message. Are you absolutely sure it will never happen? If not, I would suggest restoring this behavior.